apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/db/ScanAggregator.scala

```diff
@@ -818,39 +828,66 @@ final class ScanAggregator(
-        from      new_totals nt
-        left join previous_totals pt
-        on        nt.party = pt.party
+      previous_totals as (
+        select  rpt.party as party,
+                coalesce(cumulative_app_rewards, 0) as prev_cumulative_app_rewards,
+                coalesce(cumulative_validator_rewards, 0) as prev_cumulative_validator_rewards,
+                coalesce(cumulative_traffic_purchased, 0) as prev_cumulative_traffic_purchased,
+                coalesce(cumulative_traffic_purchased_cc_spent, 0) as prev_cumulative_traffic_purchased_cc_spent,
+                coalesce(cumulative_traffic_num_purchases, 0) as prev_cumulative_traffic_num_purchases,
+                coalesce(cumulative_change_to_initial_amount_as_of_round_zero, 0) as prev_cumulative_change_to_initial_amount_as_of_round_zero,
+                coalesce(cumulative_change_to_holding_fees_rate, 0) as prev_cumulative_change_to_holding_fees_rate
+        from    round_party_totals rpt
+        join    active_parties ap
+        on      rpt.store_id = ap.store_id
+        and     rpt.party = ap.party
+        and     rpt.closed_round = ap.closed_round
+        where   rpt.store_id = $roundTotalsStoreId
       )
+      select    nt.round,
+                nt.party,
+                nt.app_rewards,
+                nt.validator_rewards,
+                nt.traffic_purchased,
+                nt.traffic_purchased_cc_spent,
+                nt.traffic_num_purchases,
+                sum(nt.app_rewards) over (partition by nt.party order by round) + coalesce(pt.prev_cumulative_app_rewards,0) as cumulative_app_rewards,
+                sum(nt.validator_rewards) over (partition by nt.party order by round) + coalesce(pt.prev_cumulative_validator_rewards,0) as cumulative_validator_rewards,
+                sum(nt.traffic_purchased) over (partition by nt.party order by round) + coalesce(pt.prev_cumulative_traffic_purchased,0) as cumulative_traffic_purchased,
+                sum(nt.traffic_purchased_cc_spent) over (partition by nt.party order by round) + coalesce(pt.prev_cumulative_traffic_purchased_cc_spent,0) as cumulative_traffic_purchased_cc_spent,
+                sum(nt.traffic_num_purchases) over (partition by nt.party order by round) + coalesce(pt.prev_cumulative_traffic_num_purchases,0) as cumulative_traffic_num_purchases,
+                sum(nt.change_to_initial_amount_as_of_round_zero) over (partition by nt.party order by round) + coalesce(pt.prev_cumulative_change_to_initial_amount_as_of_round_zero,0) as cumulative_change_to_initial_amount_as_of_round_zero,
+                sum(nt.change_to_holding_fees_rate) over (partition by nt.party order by round) + coalesce(pt.prev_cumulative_change_to_holding_fees_rate,0) as cumulative_change_to_holding_fees_rate
+      from      new_totals nt
+      left join previous_totals pt
+      on        nt.party = pt.party;
+
+      insert into active_parties (
```

Comment:
We have an insert here and then an update below. Why do we need both? Aren't you just doing an upsert i.e. insert the max round for each party from temp_cumulative_totals and `on conflict do update round = max_round`?

Reply:
I read around and convinced myself that it was faster to insert select only what is not there. Most of the time it's updates and conflicting on insert most of the time seemed like a bad idea. But on another note I see now that I have a strange leftover here!

Reply:
But maybe I am wrong about this assumption and upserts might be better, I'll check this again, thanks for spotting this.